src/uint/modular/runtime_mod.rs

```diff
@@ -59,6 +71,33 @@ impl<const LIMBS: usize> DynResidueParams<LIMBS> {
         }
     }
 
+    /// Instantiates a new set of `ResidueParams` representing the given `modulus` if it is odd.
+    /// Returns a `CtOption` that is `None` if the provided modulus is not odd; this is a safer version of [`new`][`DynResidueParams::new`], which can panic.
+    pub fn new_checked(modulus: &Uint<LIMBS>) -> CtOption<Self> {
```

Comment:
Could this just call `new()`, so that we don't have to repeat all the parameter-generating code?

Reply:
I don't think one constructor can neatly wrap the other, but I just pushed a commit that has both constructors use a common helper function to do the heavy lifting. Let me know what you think.

Reply:
I think this is fine, we will be removing `new_checked()` in 0.6 anyway.

Reply:
Yeah exactly, a const constructor for a modulus which is intended to vary at runtime is a bit pointless.

We can just ditch it in the next release and rename `new_checked` to `new`